sc-poc/sources/openapi/helpers.go

```diff
@@ -0,0 +1,241 @@
+package openapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/spacecloud-io/space-cloud/modules/rpc"
+	"github.com/spacecloud-io/space-cloud/pkg/apis/core/v1alpha1"
+)
+
+// createRPC creates a RPC from an OpenAPI operation
+func (s *Source) createRPC(url, operationID, operationType, method string, respBody, reqBody *openapi3.Schema, parameters openapi3.Parameters, plugins []v1alpha1.HTTPPlugin) *rpc.RPC {
+	if operationType == "" {
+		operationType = "mutation"
+		if method == http.MethodGet {
+			operationType = "query"
+		}
+	}
+	for _, param := range parameters {
+		if reqBody == nil {
+			reqBody = &openapi3.Schema{
+				Type:       "object",
+				Properties: make(openapi3.Schemas),
+				Required:   make([]string, 0),
+			}
+		}
+		reqBody.Properties[param.Value.Name] = param.Value.Schema
+		if param.Value.Required {
+			reqBody.Required = append(reqBody.Required, param.Value.Name)
+		}
+
+		contentSchema, ok := param.Value.Content["application/json"]
+		if !ok {
+			continue
+		}
+		if contentSchema.Schema != nil {
+			reqBody.Properties[param.Value.Name] = contentSchema.Schema
+		}
+
+	}
+	var reqSchema *openapi3.SchemaRef
+	var respSchema *openapi3.SchemaRef
+	if reqBody != nil {
+		reqSchema = &openapi3.SchemaRef{Value: reqBody}
+	}
+	if respBody != nil {
+		respSchema = &openapi3.SchemaRef{Value: respBody}
+	}
+
+	cleanseSchemaRef(reqSchema)
+	cleanseSchemaRef(respSchema)
+	return &rpc.RPC{
+		Name:           operationID,
+		OperationType:  operationType,
+		RequestSchema:  reqSchema,
+		ResponseSchema: respSchema,
+		HTTPOptions: &v1alpha1.HTTPOptions{
+			Method: method,
+		},
```

Comment:
No need for this

Reply:
We need this or else we all the rpc will be exposed as POST endpoint. Is that the expected behavior?

Reply:
Yes